src/script.c

```diff
@@ -933,27 +944,27 @@ int wally_varbuff_to_bytes(const unsigned char *bytes, size_t bytes_len,
     return WALLY_OK;
 }
 
-int wally_witness_program_from_bytes(const unsigned char *bytes, size_t bytes_len,
-                                     uint32_t flags,
-                                     unsigned char *bytes_out, size_t len, size_t *written)
+int wally_witness_program_from_bytes_and_version(const unsigned char *bytes, size_t bytes_len,
+                                                 uint32_t version, uint32_t flags,
+                                                 unsigned char *bytes_out, size_t len,
+                                                 size_t *written)
 {
     int ret;
     unsigned char *p = bytes_out;
 
     if (written)
         *written = 0;
 
-    if ((bytes_len && !bytes) ||
+    if ((bytes_len && !bytes) || version > 16u ||
         !script_flags_ok(flags, WALLY_SCRIPT_AS_PUSH) ||
         !bytes_out || !len || !written)
         return WALLY_EINVAL;
 
     if (flags & ALL_SCRIPT_HASH_FLAGS) {
         if (!bytes_len)
             return WALLY_EINVAL;
-    } else if (bytes_len != HASH160_LEN && bytes_len != SHA256_LEN) {
-        /* Only v0 witness scripts are currently supported */
-        return WALLY_EINVAL;
+    } else if (version == 0 && bytes_len != HASH160_LEN && bytes_len != SHA256_LEN) {
+        return WALLY_EINVAL; /* Invalid length for v0 witness script */
```

Comment:
Should we check that `bytes_len` is less than or equal to 40 if version is not 0?